### PR TITLE
docs/observability: document how to configure metrics scraping

### PR DESCRIPTION
**Pull request by shashankram**
Adds doumentation for metrics scraping configuration.

Part of #1016

Signed-off-by: Shashank Ram <shashank08@gmail.com>

- New Functionality      [ ]
- Documentation          [X]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
`No`

### CODE REVIEW

File: docs/patterns/observability.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ OSM is able to automatically provision Prometheus and Grafana instances to monit
 
 ### Automatic bring up
 
-By default, OSM installation will deploy a Prometheus instance to scrape the sidecar metrics' endpoints. OSM will annotate the pods joined in the mesh with necessary sections to later have Prometheus reach and scrape the pods, also by default. To install a Grafana stack, set the enable-grafana flag to true when doing OSM install. 
+By default, OSM installation will deploy a Prometheus instance to scrape the sidecar's metrics endpoints. Based on the metrics scraping configuration set by the user, OSM will annotate pods part of the mesh with necessary metrics annotations to have Prometheus reach and scrape the pods to collect relevant metrics. To install Grafana for metrics visualization, set the `enable-grafana` flag to true when installing OSM using the `osm install` command.
```
Comment (snehachhabria) [inline docs/patterns/observability.md:17]:
```suggestion
By default, OSM installation will deploy a Prometheus instance to scrape the sidecar's metrics endpoints. Based on the metrics scraping configuration set by the user, OSM will annotate pods that are a part of the mesh with necessary metrics annotations enabling Prometheus to reach and scrape the pods for relevant metrics. To install Grafana for metrics visualization, set the `enable-grafana` flag to true when installing OSM using the `osm install` command.
```

File: docs/patterns/observability.md [hunk 2]
```diff
@@ -109,6 +109,26 @@ pushing the listener configuration to the pods for Prometheus to reach:
       target_label: source_workload_name
 ```
 
+## Configuring Prometheus metrics scraping
+Metrics scraping can be configured using the `osm metrics` command. By default, OSM does not configure metrics scraping for pods in the mesh. Metrics scraping can be enabled or disabled at namespace scope such that pods belonging to configured namespaces can be enabled or disabled for scraping metrics.
```
Comment (snehachhabria) [inline docs/patterns/observability.md:113]:
```suggestion
Metrics scraping can be configured using the `osm metrics` command. By default, OSM **does not** configure metrics scraping for pods in the mesh. Metrics scraping can be enabled or disabled at namespace scope such that pods belonging to configured namespaces can be enabled or disabled for scraping metrics.
```

File: docs/patterns/observability.md [hunk 3]
```diff
@@ -109,6 +109,26 @@ pushing the listener configuration to the pods for Prometheus to reach:
...
+
+For metrics to be scraped, the following prerequisites must be met:
+- A Prometheus instance must be deployed either using [automatic bringup of Prometheus](#automatic-bring-up) or using the [bring your own Prometheus guide](#byo-bring-your-own).
```
Comment (eduser25) [inline docs/patterns/observability.md:116]: I'd relax the wording a bit here
```suggestion
- A running service able to scrap Prometheus endpoints. OSM provides configuration for an [automatic bringup of Prometheus](#automatic-bring-up); alternatively users can [bring their own Prometheus](#byo-bring-your-own).
```

**Review by eduser25 — APPROVED** [review]

**Review by eduser25 — APPROVED** [review]

File: docs/patterns/observability.md (hunk 1 above)
Comment (shashankram) [reply docs/patterns/observability.md:17]: Existing grammar seems fine to me.

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**